### PR TITLE
Feature/sfat 251 - Agreements Service BaT extensions

### PR DESCRIPTION
**Pull request by adrianmilne**

### CODE REVIEW

File: src/main/java/uk/gov/crowncommercial/dts/scale/service/agreements/converter/AgreementConverter.java [hunk 1]
```diff
@@ -6,23 +6,32 @@
...
 @Component
 public class AgreementConverter {
 
   @Autowired
   private ModelMapper modelMapper;
 
+  private final AgreementService service;
+
   @PostConstruct
   public void init() {
```
Comment (tommyb82) [inline src/main/java/uk/gov/crowncommercial/dts/scale/service/agreements/converter/AgreementConverter.java:36]: SonarLint is starting to complain about the complexity of this method.. I don't see it as an issue though, it's pretty clear what's going on (just one to keep in mind for future extensions)

File: src/main/java/uk/gov/crowncommercial/dts/scale/service/agreements/model/entity/LotRelatedLot.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+import org.hibernate.annotations.Immutable;
+import lombok.AccessLevel;
+import lombok.Data;
+import lombok.experimental.FieldDefaults;
+
+@Entity
+@Immutable
+@Table(name = "lot_related_lots")
```
Comment (tommyb82) [inline src/main/java/uk/gov/crowncommercial/dts/scale/service/agreements/model/entity/LotRelatedLot.java:17]: Not quite sure about the table naming but not an issue for here..

File: src/main/java/uk/gov/crowncommercial/dts/scale/service/agreements/converter/AgreementConverter.java (hunk 1 above)
Comment (adrianmilne) [reply src/main/java/uk/gov/crowncommercial/dts/scale/service/agreements/converter/AgreementConverter.java:36]: Ah yes - didn't spot that. I can pull the converters out into separate classes - should get rid of that - I'll have a look

Comment (adrianmilne) [reply src/main/java/uk/gov/crowncommercial/dts/scale/service/agreements/converter/AgreementConverter.java:31]: I've refactored those out now

File: src/main/java/uk/gov/crowncommercial/dts/scale/service/agreements/model/entity/LotRelatedLot.java (hunk 2 above)
Comment (adrianmilne) [reply src/main/java/uk/gov/crowncommercial/dts/scale/service/agreements/model/entity/LotRelatedLot.java:17]: Yep - found that quite confusing at the time as it's linking via Lot Rule

**Pull request merged**